rfcs/00x-frontend-library-split.md

```diff
@@ -0,0 +1,88 @@
+- Feature Name: frontend-library-split
+- Start Date: 2020-11-23
+- RFC PR: [amundsen-io/rfcs#150(https://github.com/amundsen-io/rfcs/pull/15) (after opening the RFC PR, update this with a link to it and update the file name)
+- Amundsen Issue: [amundsen-io/amundsen#0000](https://github.com/amundsen-io/amundsen/issues/0000) (leave this empty for now)
+
+# Split of the frontendlibrary repository in two
+
+## Summary
+
+At this moment, our frontend repository contains a Flask application that acts as an API server and also serves a frontend application based in React.js. This means that there are two applications (Flask and Frontend), with two languages (Python and JavaScript), two configuration settings, two dependency management systems (pip and npm), two sets of owners (frontend and backend developers), and two main "responsibilities" (API server and User Interface application) baked into the same repository. This is a problem, as it makes the system complex, couples both systems and stifles the evolution of both applications.
```

Comment:
Currently Amundsen includes three services, the FE is just one of them. If we split, any reason we need the API server instead of calling metadata/search directly.

Reply:
That's a great question, maybe we could remove this step and call metadata/search directly.
I see it as a good "proxy" server for the functionality on those services, and I guess it would be great if we could make it extensible through plugins, so we could implement things like the emailing feature we have at Lyft.

Reply:
I see the potential value of the API server as a proxy. Are there other use cases we know of that would benefit from a proxy?